dbms/src/IO/WriteBufferFromVector.h

```diff
@@ -57,7 +57,7 @@ class WriteBufferFromVector : public WriteBuffer
         : WriteBuffer(nullptr, 0), vector(vector_)
     {
         size_t old_size = vector.size();
-        vector.resize(std::max(vector.size() + initial_size, vector.capacity()));
+        vector.resize(std::max(initial_size, vector.capacity()));
```

Comment:
The change in `WriteBuffer::next()` fixes the hanging. We can still add `initial_size` to `vector.size()` here but it seems it's better not to do it in this constructor in advance because `WriteBuffer::nextImpl()` also can increase the size of the vector.

Reply:
This commend isn't actual anymore. Here I only corrected the formula for consistency with `WriteBufferFromVector::nextImpl()`.